easypost/shipment.py

```diff
@@ -62,24 +66,46 @@ def label(self, **params) -> "Shipment":
 
     def lowest_rate(self, carriers: List[str] = None, services: List[str] = None) -> Rate:
         """Get the lowest rate of a shipment."""
-        carriers = carriers or []
-        services = services or []
-        lowest_rate = None
+        lowest_rate = get_lowest_object_rate(self, carriers, services)
+
+        return lowest_rate
 
-        carriers = [c.lower() for c in carriers]
-        services = [service.lower() for service in services]
+    def lowest_smartrate(self, delivery_days: Optional[int] = None, delivery_accuracy: Optional[str] = None) -> Rate:
+        """Get the lowest smartrate of a shipment."""
+        smartrates = self.get_smartrates()
+        lowest_smartrate = self.get_lowest_smartrate(smartrates, delivery_days, delivery_accuracy)
 
-        for rate in self.rates:
-            if len(carriers) > 0 and rate.carrier.lower() not in carriers:
-                continue
+        return lowest_smartrate
 
-            if len(services) > 0 and rate.service.lower() not in services:
-                continue
+    @staticmethod
+    def get_lowest_smartrate(
+        smartrates, delivery_days: Optional[int] = None, delivery_accuracy: Optional[str] = None
+    ) -> Rate:
+        """Get the lowest smartrate from a list of smartrates."""
+        valid_delivery_accuracy_values = {
+            "percentile_50",
+            "percentile_75",
+            "percentile_85",
+            "percentile_90",
+            "percentile_95",
+            "percentile_97",
+            "percentile_99",
+        }
+        lowest_smartrate = None
 
-            if lowest_rate is None or float(rate.rate) < float(lowest_rate.rate):
-                lowest_rate = rate
+        for rate in smartrates:
+            if delivery_days and delivery_accuracy:
+                if delivery_accuracy not in valid_delivery_accuracy_values:
+                    raise Error(
+                        message=f"Invalid delivery_accuracy value, must be one of: {valid_delivery_accuracy_values}"
+                    )
```

Comment:
I would move these two if statements before the loop since we don't need to check them every time inside the loop.